runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -178,17 +205,18 @@ void registerTable(@Nonnull String namespace,
         K defaultKeyMessage = (K) keyClass.getMethod("getDefaultInstance").invoke(null);
         V defaultValueMessage = (V) payloadClass.getMethod("getDefaultInstance").invoke(null);
 
+        Map<ProtobufFileName, CorfuRecord<ProtobufFileDescriptor, TableMetadata>> allDescriptors = new HashMap<>();
```

Comment:
This will only de-duplicate the descriptors within a single table, right? But wouldn't you also want to de-duplicate across tables?

Reply:
we do deduplicate across tables, please see the protobufDescriptorTable where the key is protobuf file name and value is the descriptor for that file and that is shared across all tables in the table registry.